src/openkit-shared/Core/Action.cs

```diff
@@ -204,5 +232,11 @@ public void Dispose()
         {
             LeaveAction();
         }
+
+        public override string ToString()
+        {
+            return GetType().Name + " [sn=" + beacon.SessionNumber + ", id=" + id + ", name=" + name + ", pa="

```

Comment:
Just as information (not necessary to change):
String interpolation would work here as well.